homeassistant/components/color_extractor/__init__.py

```diff
@@ -0,0 +1,157 @@
+"""Module for color_extractor (RGB extraction from images) component."""
+import asyncio
+import io
+import logging
+
+from PIL import UnidentifiedImageError
+import aiohttp
+import async_timeout
+from colorthief import ColorThief
+import voluptuous as vol
+
+from homeassistant.components.color_extractor.const import (
+    ATTR_PATH,
+    ATTR_URL,
+    DOMAIN,
+    SERVICE_TURN_ON,
+)
+from homeassistant.components.light import (
+    ATTR_RGB_COLOR,
+    DOMAIN as LIGHT_DOMAIN,
+    LIGHT_TURN_ON_SCHEMA,
+    SERVICE_TURN_ON as LIGHT_SERVICE_TURN_ON,
+)
+from homeassistant.helpers import aiohttp_client
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+# Extend the existing light.turn_on service schema
+SERVICE_SCHEMA = cv.make_entity_service_schema(
+    {
+        **LIGHT_TURN_ON_SCHEMA,
+        vol.Exclusive(ATTR_PATH, "color_extractor"): cv.isfile,
+        vol.Exclusive(ATTR_URL, "color_extractor"): cv.url,
+    }
+)
+
+
+def _get_file(file_path):
+    """Get a PIL acceptable input file reference.
+
+    Allows us to mock patch during testing to make BytesIO stream.
+    """
+    return file_path
+
+
+async def async_setup(hass, hass_config):
+    """Set up services for color_extractor integration."""
+
+    def _get_color(file_handler) -> tuple:
+        """Given an image file, extract the predominant color from it."""
+        color_thief = ColorThief(file_handler)
+
+        # get_color returns a SINGLE RGB value for the given image
+        color = color_thief.get_color(quality=1)
+
+        _LOGGER.debug("Extracted RGB color %s from image", color)
+
+        return color
+
+    async def async_handle_service(service_call):
+        """Decide which color_extractor method to call based on service."""
+        service_data = dict(service_call.data)
+
+        if ATTR_URL in service_data:
+            await _async_url_service(service_data)
+        elif ATTR_PATH in service_data:
+            await _async_file_service(service_data)
+        else:
+            _LOGGER.error("Missing either required url or path key")
+            return None
```

Comment:
There is no return value for service handlers.
```suggestion
            return
```